docs/pages/admin-guides/infrastructure-as-code/terraform-starter/rbac.mdx

```diff
@@ -0,0 +1,549 @@
+     request_roles = [module.prod_role.role_name]
+   }
+   ```
+   </TabItem>
+   </Tabs>
+
+Next, we will show you how to configure the two child modules, and walk you
+through the Terraform resources that they apply.
+
+## Step 2/4. Configure role principals
+
+Together, the `prod_role` and `dev_role` modules you declared in Step 1 create
+three Teleport roles:
+
+|Role|Description|
+|---|---|
+|`prod_access`|Allows access to infrastructure resources with the `env:prod` label.|
+|`dev_access`|Allows access to infrastructure resources with the `env:dev` label, and Access Requests for the `prod_access` role.|
+|`prod_reviewer`|Allows reviews of Access Requests for the `prod_access` role.|
+
+When Teleport users connect to resources in your infrastructure, they assume a
+principal, such as an operating system login or Kubernetes user, in order to
+interact with those resources. In this step, you will configure the `prod_role`
+and `dev_role` modules to grant access to principals in your infrastructure.
+
+In `rbac.tf`, edit the `prod_role` and `dev_role` blocks so that the
+`principals` field contains a mapping, similar to the example below. Use the
+list of keys below the example to configure the principals.
+
+```hcl
+module "prod_role" {
+  principals = {
+    KEY = "value"
+  }
+  // ...
+}
+
+// ...
+```
+
+|Key|Description|
+|---|---|
+|`aws_role_arns`|AWS role ARNs the user can access when authenticating to an AWS API.|
+|`azure_identities`|Azure identities the user can access when authenticating to an Azure API.|
+|`db_names`|Names of databases the user can access within a database server.|
+|`db_roles`|Roles the user can access on a database when they authenticate to a database server.|
+|`db_users`|Users the user can access on a database when they authenticate to a database server.|
+|`gcp_service_accounts`|Google Cloud service accounts the user can access when authenticating to a Google Cloud API.|
+|`kubernetes_groups`|Kubernetes groups the Teleport Database Service can impersonate when proxying requests from the user.|
+|`kubernetes_users`|Kubernetes users the Teleport Database Service can impersonate when proxying requests from the user.|
+|`logins`|Operating system logins the user can access when authenticating to a Linux server.|
+|`windows_desktop_logins`|Operating system logins the user can access when authenticating to a Windows desktop.|
+
+For example, the following configuration allows users with the `dev_access` role
+to assume the `dev` login on Linux hosts and the `developers` group on
+Kubernetes. `prod` users have more privileges and can assume the `root` login
+and `system:masters` Kubernetes group:
+
+```hcl
+module "dev_role" {
+  principals = {
+    logins = ["dev"]
+    kubernetes_groups = ["developers"]
+  }
+  // ...
+}
+
+module "prod_role" {
+  principals = {
+    logins = ["root"]
+    kubernetes_groups = ["system:masters"]
+  }
+  // ...
+}
+```
+
+## Step 3/4. Configure the single sign-on connector
```

Comment:
Is there a good way to make this part optional? For a quick getting started guide, adding in SAML/OIDC can complicate it fast and add potential roadblocks if it doesn't go well. And as a user, I might not have access to a test/demo SSO provider to do experiments with and get stuck because of that.

Reply:
Overall the guide looks great!